src/python/pants/backend/python/lint/pyupgrade/subsystem.py

```diff
@@ -0,0 +1,75 @@
+# Copyright 2021 Pants project contributors (see CONTRIBUTORS.md).
+# Licensed under the Apache License, Version 2.0 (see LICENSE).
+
+from __future__ import annotations
+
+from typing import cast
+
+from pants.backend.python.goals.lockfile import PythonLockfileRequest, PythonToolLockfileSentinel
+from pants.backend.python.subsystems.python_tool_base import PythonToolBase
+from pants.backend.python.target_types import ConsoleScript
+from pants.engine.rules import collect_rules, rule
+from pants.engine.unions import UnionRule
+from pants.option.custom_types import shell_str
+from pants.util.docutil import git_url
+
+
+class PyUpgrade(PythonToolBase):
+    options_scope = "pyupgrade"
+    help = (
+        "Upgrade syntax for newer versions of the language (https://github.com/asottile/pyupgrade)."
+    )
+
+    default_version = "pyupgrade==2.29.0"
+    default_main = ConsoleScript("pyupgrade")
+
+    register_interpreter_constraints = True
+    default_interpreter_constraints = ["CPython>=3.7"]
+
+    register_lockfile = True
+    default_lockfile_resource = ("pants.backend.python.lint.pyupgrade", "lockfile.txt")
+    default_lockfile_path = "src/python/pants/backend/python/lint/pyupgrade/lockfile.txt"
+    default_lockfile_url = git_url(default_lockfile_path)
+
+    @classmethod
+    def register_options(cls, register):
+        super().register_options(register)
+        register(
+            "--skip",
+            type=bool,
+            default=False,
+            help=(
+                f"Don't use pyupgrade when running `{register.bootstrap.pants_bin_name} fmt` and "
+                f"`{register.bootstrap.pants_bin_name} lint`."
+            ),
+        )
+        register(
+            "--args",
+            type=list,
+            default=["--py36-plus"],
```

Comment:
I at first felt uncomfortable setting this by default, but it looks like you can pass `pyupgrade --py36-plus --py37-upgrade` at the same time, so I'm a little more comfortable with it..Seems like a sensible default.

Reply:
I fully expect anyone who uses this tool to override this default.

Reply:
Hm, then I recommend removing this. Note that specifying `--pyupgrade-args='--py37-plus'` on the CLI _appends_ to the option, rather than overriding. You would have to use `"['--py37-plus']"` to override, which is clunky.

This is why I was nervous about having this be the default. It's awkward in Pants to remove a single element from a list option: you have to override the option entirely. 

I think I recommend removing the `default`..and then we update the docs to point out that you almost certainly will want to set `[pyupgrade].args` in your `pants.toml`.